runtime/compiler/env/VMJ9Server.cpp

```diff
@@ -1015,6 +1064,43 @@ TR_J9ServerVM::sameClassLoaders(TR_OpaqueClassBlock * class1, TR_OpaqueClassBloc
 bool
 TR_J9ServerVM::isUnloadAssumptionRequired(TR_OpaqueClassBlock *clazz, TR_ResolvedMethod *methodBeingCompiled)
    {
+   TR_OpaqueClassBlock * classOfMethod = static_cast<TR_ResolvedJ9JITaaSServerMethod *>(methodBeingCompiled)->classOfMethod();
+   if (clazz == classOfMethod)
+      {
+      return false;
+      }
+
+      {
+      OMR::CriticalSection getRemoteROMClass(_compInfoPT->getClientData()->getROMMapMonitor());
+      auto it_class = _compInfoPT->getClientData()->getROMClassMap().find((J9Class*) clazz);
+      if (it_class != _compInfoPT->getClientData()->getROMClassMap().end())
+         {
+         if (it_class->second.classLoader == getSystemClassLoader())
+            {
+            return false;
+            }
+         if (!isAnonymousClass(it_class->second.romClass))
+            {
+            auto it_classOfMethod = _compInfoPT->getClientData()->getROMClassMap().find((J9Class*) classOfMethod);
+            if (it_classOfMethod != _compInfoPT->getClientData()->getROMClassMap().end())
+               {
+               if (it_class->second.classLoader == it_classOfMethod->second.classLoader)
+                  {
+                  return false;
+                  }
+               else
+                  {
+                  return true;
+                  }
```

Comment:
How about `return it_class->second.classLoader != it_classOfMethod->second.classLoader` to cut 8 lines into 1.